base/MAPL_ISO8601_Time.F90

```diff
@@ -0,0 +1,720 @@
+!Allow 60 for seconds
+!No fractions other than milliseconds (explicit)
+!No weeks
+!Z required for time
+!No parts of day !Allow ordinal dates as input/output
+#include "MAPL_Exceptions.h"
+#include "MAPL_ErrLog.h"
+module MAPL_ISO8601_Time
+   use MAPL_KeywordEnforcerMod
+   use MAPL_ExceptionHandling
+   use ESMF
+   implicit none
+
+!   private
+   public :: DD
+   public :: TD
+   public :: TP
+   public :: TZ
```

Comment:
If these really need to be PUBLIC (hope not), then they will need long informative names.   `ASSOCIATE` can be used to give them shorter names when actually used in expressions.